### PR TITLE
Add rita-analyzer example

### PR DESCRIPTION
**Pull request by SevaMorotskiy**

### CODE REVIEW

**Comment by tzach** [comment]
Failed to compile the example. Error at the bottom.

```
sbt assembly
Getting org.scala-sbt sbt 0.13.13 ...
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/sbt/0.13.13/jars/sbt.jar ...
	[SUCCESSFUL ] org.scala-sbt#sbt;0.13.13!sbt.jar (2393ms)
downloading https://repo1.maven.org/maven2/org/scala-lang/scala-library/2.10.6/scala-library-2.10.6.jar ...
	[SUCCESSFUL ] org.scala-lang#scala-library;2.10.6!scala-library.jar (1482ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/main/0.13.13/jars/main.jar ...
	[SUCCESSFUL ] org.scala-sbt#main;0.13.13!main.jar (5155ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/compiler-interface/0.13.13/jars/compiler-interface.jar ...
	[SUCCESSFUL ] org.scala-sbt#compiler-interface;0.13.13!compiler-interface.jar (3481ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/actions/0.13.13/jars/actions.jar ...
	[SUCCESSFUL ] org.scala-sbt#actions;0.13.13!actions.jar (2499ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/main-settings/0.13.13/jars/main-settings.jar ...
	[SUCCESSFUL ] org.scala-sbt#main-settings;0.13.13!main-settings.jar (2702ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/interface/0.13.13/jars/interface.jar ...
	[SUCCESSFUL ] org.scala-sbt#interface;0.13.13!interface.jar (2236ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/io/0.13.13/jars/io.jar ...
	[SUCCESSFUL ] org.scala-sbt#io;0.13.13!io.jar (3740ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/ivy/0.13.13/jars/ivy.jar ...
	[SUCCESSFUL ] org.scala-sbt#ivy;0.13.13!ivy.jar (3061ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/logging/0.13.13/jars/logging.jar ...
	[SUCCESSFUL ] org.scala-sbt#logging;0.13.13!logging.jar (2286ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/logic/0.13.13/jars/logic.jar ...
	[SUCCESSFUL ] org.scala-sbt#logic;0.13.13!logic.jar (2262ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/process/0.13.13/jars/process.jar ...
	[SUCCESSFUL ] org.scala-sbt#process;0.13.13!process.jar (2339ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/run/0.13.13/jars/run.jar ...
	[SUCCESSFUL ] org.scala-sbt#run;0.13.13!run.jar (2554ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/command/0.13.13/jars/command.jar ...
	[SUCCESSFUL ] org.scala-sbt#command;0.13.13!command.jar (4810ms)
downloading https://repo1.maven.org/maven2/org/scala-sbt/launcher-interface/1.0.0-M1/launcher-interface-1.0.0-M1.jar ...
	[SUCCESSFUL ] org.scala-sbt#launcher-interface;1.0.0-M1!launcher-interface.jar (541ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/classpath/0.13.13/jars/classpath.jar ...
	[SUCCESSFUL ] org.scala-sbt#classpath;0.13.13!classpath.jar (2261ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/completion/0.13.13/jars/completion.jar ...
	[SUCCESSFUL ] org.scala-sbt#completion;0.13.13!completion.jar (14860ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/api/0.13.13/jars/api.jar ...
	[SUCCESSFUL ] org.scala-sbt#api;0.13.13!api.jar (2550ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/compiler-integration/0.13.13/jars/compiler-integration.jar ...
	[SUCCESSFUL ] org.scala-sbt#compiler-integration;0.13.13!compiler-integration.jar (2437ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/compiler-ivy-integration/0.13.13/jars/compiler-ivy-integration.jar ...
	[SUCCESSFUL ] org.scala-sbt#compiler-ivy-integration;0.13.13!compiler-ivy-integration.jar (2350ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/relation/0.13.13/jars/relation.jar ...
	[SUCCESSFUL ] org.scala-sbt#relation;0.13.13!relation.jar (2196ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/task-system/0.13.13/jars/task-system.jar ...
	[SUCCESSFUL ] org.scala-sbt#task-system;0.13.13!task-system.jar (2335ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/tasks/0.13.13/jars/tasks.jar ...
	[SUCCESSFUL ] org.scala-sbt#tasks;0.13.13!tasks.jar (2330ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/tracking/0.13.13/jars/tracking.jar ...
	[SUCCESSFUL ] org.scala-sbt#tracking;0.13.13!tracking.jar (2246ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/testing/0.13.13/jars/testing.jar ...
	[SUCCESSFUL ] org.scala-sbt#testing;0.13.13!testing.jar (2299ms)
downloading https://repo1.maven.org/maven2/org/scala-lang/scala-compiler/2.10.6/scala-compiler-2.10.6.jar ...
	[SUCCESSFUL ] org.scala-lang#scala-compiler;2.10.6!scala-compiler.jar (1767ms)
downloading https://repo1.maven.org/maven2/org/scala-lang/scala-reflect/2.10.6/scala-reflect-2.10.6.jar ...
	[SUCCESSFUL ] org.scala-lang#scala-reflect;2.10.6!scala-reflect.jar (535ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/control/0.13.13/jars/control.jar ...
	[SUCCESSFUL ] org.scala-sbt#control;0.13.13!control.jar (2131ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/collections/0.13.13/jars/collections.jar ...
	[SUCCESSFUL ] org.scala-sbt#collections;0.13.13!collections.jar (4482ms)
downloading https://repo1.maven.org/maven2/jline/jline/2.13/jline-2.13.jar ...
	[SUCCESSFUL ] jline#jline;2.13!jline.jar (800ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/classfile/0.13.13/jars/classfile.jar ...
	[SUCCESSFUL ] org.scala-sbt#classfile;0.13.13!classfile.jar (2205ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/incremental-compiler/0.13.13/jars/incremental-compiler.jar ...
	[SUCCESSFUL ] org.scala-sbt#incremental-compiler;0.13.13!incremental-compiler.jar (4395ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/compile/0.13.13/jars/compile.jar ...
	[SUCCESSFUL ] org.scala-sbt#compile;0.13.13!compile.jar (3357ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/persist/0.13.13/jars/persist.jar ...
	[SUCCESSFUL ] org.scala-sbt#persist;0.13.13!persist.jar (2322ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/cross/0.13.13/jars/cross.jar ...
	[SUCCESSFUL ] org.scala-sbt#cross;0.13.13!cross.jar (2080ms)
downloading https://repo1.maven.org/maven2/org/scala-sbt/ivy/ivy/2.3.0-sbt-2cf13e211b2cb31f0d3b317289dca70eca3362f6/ivy-2.3.0-sbt-2cf13e211b2cb31f0d3b317289dca70eca3362f6.jar ...
	[SUCCESSFUL ] org.scala-sbt.ivy#ivy;2.3.0-sbt-2cf13e211b2cb31f0d3b317289dca70eca3362f6!ivy.jar (939ms)
downloading https://repo1.maven.org/maven2/org/scala-sbt/serialization_2.10/0.1.2/serialization_2.10-0.1.2.jar ...
	[SUCCESSFUL ] org.scala-sbt#serialization_2.10;0.1.2!serialization_2.10.jar (329ms)
downloading https://repo1.maven.org/maven2/org/scala-lang/modules/scala-pickling_2.10/0.10.1/scala-pickling_2.10-0.10.1.jar ...
	[SUCCESSFUL ] org.scala-lang.modules#scala-pickling_2.10;0.10.1!scala-pickling_2.10.jar (350ms)
downloading https://repo1.maven.org/maven2/org/scalamacros/quasiquotes_2.10/2.0.1/quasiquotes_2.10-2.0.1.jar ...
	[SUCCESSFUL ] org.scalamacros#quasiquotes_2.10;2.0.1!quasiquotes_2.10.jar (366ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/cache/0.13.13/jars/cache.jar ...
	[SUCCESSFUL ] org.scala-sbt#cache;0.13.13!cache.jar (2314ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/test-agent/0.13.13/jars/test-agent.jar ...
	[SUCCESSFUL ] org.scala-sbt#test-agent;0.13.13!test-agent.jar (2167ms)
downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/apply-macro/0.13.13/jars/apply-macro.jar ...
	[SUCCESSFUL ] org.scala-sbt#apply-macro;0.13.13!apply-macro.jar (2212ms)
downloading https://repo1.maven.org/maven2/org/scala-sbt/template-resolver/0.1/template-resolver-0.1.jar ...
	[SUCCESSFUL ] org.scala-sbt#template-resolver;0.1!template-resolver.jar (538ms)
:: retrieving :: org.scala-sbt#boot-app
	confs: [default]
	52 artifacts copied, 0 already retrieved (17598kB/269ms)
Getting Scala 2.10.4 (for sbt)...
downloading https://repo1.maven.org/maven2/org/scala-lang/jline/2.10.4/jline-2.10.4.jar ...
	[SUCCESSFUL ] org.scala-lang#jline;2.10.4!jline.jar (503ms)
downloading https://repo1.maven.org/maven2/org/fusesource/jansi/jansi/1.4/jansi-1.4.jar ...
	[SUCCESSFUL ] org.fusesource.jansi#jansi;1.4!jansi.jar (388ms)
:: retrieving :: org.scala-sbt#boot-scala
	confs: [default]
	5 artifacts copied, 0 already retrieved (24459kB/515ms)
[info] Loading project definition from /home/tzach/projects/SevaMorotskiy-scylla-code-samples/rita-analyzer/project
/home/tzach/projects/SevaMorotskiy-scylla-code-samples/rita-analyzer/build.sbt:13: error: illegal start of simple expression
  "org.apache.spark" %% "spark-hive" % sparkVersion % "provided" exclude("com.google.guava", "guava"),
```

**Comment by avikivity** [comment]
@tzach, we don't yet have a license for this repository.  I think the Apache license is fine here, but it should be added before we merge pull requests, so contributors are aware of the license.

@SevaMorotskiy, are you okay with licensing your contribution using the Apache license?

**Comment by dorlaor** [comment]
@avikivity, Seva did this work for us ;)

**Comment by SevaMorotskiy** [comment]
@avikivity 
> are you okay with licensing your contribution using the Apache license
You mean Apache license for Apache Spark? Not sure, that I got it correctly.

**Comment by dorlaor** [comment]
He meant the license for this example repository

**Comment by SevaMorotskiy** [comment]
@tzach 


> Failed to compile the example. Error at the bottom.
> sbt assembly
> Getting org.scala-sbt sbt 0.13.13 ...
> downloading http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/sbt/0.13.13/jars/sbt.jar ...
> 	[SUCCESSFUL ] org.scala-sbt#sbt;0.13.13!sbt.jar (2393ms)
> downloading https://repo1.maven.org/maven2/org/scala-lang/scala-library/2.10.6/scala-library-2.10.6.jar 
>	[SUCCESSFUL ] org.scala-lang#scala-library;2.10.6!scala-library.jar (1482ms)

Just checked everything from scratch using new just installed server. All compiles and works fine. We checked it on the servers with CentOs, MacOs. Tell me please the name and version of the OS on the server you are using for the actions. Moreover, did you follow the instructions from the acrticle?

I need to know the following information from your server to sort the issue out:
$ scala -version
$ sbt about

**Comment by eyalgutkind** [comment]
Can we elaborate on the case of installing Scylla and Spark in a cluster mode?
Currently, the example looks at the installation/usage as a single node deployment for both Scylla and Spark.

**Comment by SevaMorotskiy** [comment]
@eyalgutkind 

> Can we elaborate on the case of installing Scylla and Spark in a cluster mode?

Yes, sure. How to do it  - you can find the file spark-scylla.conf in the folder rita-analyzer. Here is its content:
- spark.master local
- spark.cassandra.connection.host localhost

The first line is a Master URLs, so there could be the following insted of _localhost_:
     **local** - Run Spark locally with one worker thread (i.e. no parallelism at all).
     **local[K]** - Run Spark locally with K worker threads (ideally, set this to the number of cores on your machine).
     **local[*]** - Run Spark locally with as many worker threads as logical cores on your machine.
     **spark://HOST:PORT** - Connect to the given Spark standalone cluster master. The port must be whichever one your master is configured to use, which is 7077 by default.
     **mesos://HOST:PORT** - Connect to the given Mesos cluster. The port must be whichever one your is configured to use, which is 5050 by default. Or, for a Mesos cluster using ZooKeeper, use mesos://zk://.... To submit with --deploy-mode cluster, the HOST:PORT should be configured to connect to the MesosClusterDispatcher.
     **yarn** - Connect to a YARN cluster in client or cluster mode depending on the value of --deploy-mode. The cluster location will be found based on the HADOOP_CONF_DIR or YARN_CONF_DIR variable.
  
The second row is the address of Cassandra or ScyllaDB (instead of _localhost_)  without port, i.e. only  IP/DNS

File: rita-analyzer/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+git clone https://github.com/scylladb/scylla-code-samples.git
+cd scylla-code-samples/rita-analyzer
+sbt assembly
+```
+
+
+The next step is load previously downloaded data from /tmp/2008.csv file to ScyllaDB by executing the following command in Spark directory:
+```
+./bin/spark-submit --properties-file /path/to/rita-analyzer/spark-scylla.conf --class Loader /path/to/rita-analyzer/target/scala-2.11/rita-analyzer-assembly-1.0.jar
+```
```
Comment (tzach) [inline rita-analyzer/README.md]: does this command upload the data file? its look the same as the command below

**Comment by tzach** [comment]
@SevaMorotskiy updating Spark to 2.11 solves my build problem.
I'm still struggling with running the example.  
As far as I understand README instructions assumes Scylla is running on local host, but does not require Spark server running. Is it the case? 
I updated `spark-scylla.conf` to  `spark.cassandra.connection.host 172.17.0.2` (my Scylla server IP)
and got the following:

```
$ ~/sw/spark-2.1.0-bin-hadoop2.7/bin/spark-submit --properties-file spark-scylla.conf --class Loader target/scala-2.11/rita-analyzer-assembly-1.0.jar 
Using Spark's default log4j profile: org/apache/spark/log4j-defaults.properties
17/03/26 16:21:36 INFO SparkContext: Running Spark version 2.1.0
17/03/26 16:21:36 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
17/03/26 16:21:37 INFO SecurityManager: Changing view acls to: tzach
17/03/26 16:21:37 INFO SecurityManager: Changing modify acls to: tzach
17/03/26 16:21:37 INFO SecurityManager: Changing view acls groups to: 
17/03/26 16:21:37 INFO SecurityManager: Changing modify acls groups to: 
17/03/26 16:21:37 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(tzach); groups with view permissions: Set(); users  with modify permissions: Set(tzach); groups with modify permissions: Set()
17/03/26 16:21:37 INFO Utils: Successfully started service 'sparkDriver' on port 43589.
17/03/26 16:21:37 INFO SparkEnv: Registering MapOutputTracker
17/03/26 16:21:38 INFO SparkEnv: Registering BlockManagerMaster
17/03/26 16:21:38 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
17/03/26 16:21:38 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
17/03/26 16:21:38 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-b24fa1b2-b473-4945-b5d5-d44a92ba8539
17/03/26 16:21:38 INFO MemoryStore: MemoryStore started with capacity 366.3 MB
17/03/26 16:21:38 INFO SparkEnv: Registering OutputCommitCoordinator
17/03/26 16:21:38 INFO Utils: Successfully started service 'SparkUI' on port 4040.
17/03/26 16:21:38 INFO SparkUI: Bound SparkUI to 0.0.0.0, and started at http://10.0.0.161:4040
17/03/26 16:21:38 INFO SparkContext: Added JAR file:/home/tzach/projects/SevaMorotskiy-scylla-code-samples/rita-analyzer/target/scala-2.11/rita-analyzer-assembly-1.0.jar at spark://10.0.0.161:43589/jars/rita-analyzer-assembly-1.0.jar with timestamp 1490534498506
17/03/26 16:21:38 INFO Executor: Starting executor ID driver on host localhost
17/03/26 16:21:38 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 39807.
17/03/26 16:21:38 INFO NettyBlockTransferService: Server created on 10.0.0.161:39807
17/03/26 16:21:38 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
17/03/26 16:21:38 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 10.0.0.161, 39807, None)
17/03/26 16:21:38 INFO BlockManagerMasterEndpoint: Registering block manager 10.0.0.161:39807 with 366.3 MB RAM, BlockManagerId(driver, 10.0.0.161, 39807, None)
17/03/26 16:21:38 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 10.0.0.161, 39807, None)
17/03/26 16:21:38 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 10.0.0.161, 39807, None)
Exception in thread "main" java.io.FileNotFoundException: /tmp/2008.csv (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at scala.io.Source$.fromFile(Source.scala:91)
	at scala.io.Source$.fromFile(Source.scala:76)
	at scala.io.Source$.fromFile(Source.scala:54)
	at Loader$.delayedEndpoint$Loader$1(Loader.scala:14)
	at Loader$delayedInit$body.apply(Loader.scala:8)
	at scala.Function0$class.apply$mcV$sp(Function0.scala:34)
	at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:12)
	at scala.App$$anonfun$main$1.apply(App.scala:76)
	at scala.App$$anonfun$main$1.apply(App.scala:76)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at scala.collection.generic.TraversableForwarder$class.foreach(TraversableForwarder.scala:35)
	at scala.App$class.main(App.scala:76)
	at Loader$.main(Loader.scala:8)
	at Loader.main(Loader.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.deploy.SparkSubmit$.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:738)
	at org.apache.spark.deploy.SparkSubmit$.doRunMain$1(SparkSubmit.scala:187)
	at org.apache.spark.deploy.SparkSubmit$.submit(SparkSubmit.scala:212)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:126)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
```

File: rita-analyzer/README.md (hunk 1 above)
Comment (SevaMorotskiy) [inline rita-analyzer/README.md]: Yes. It is the data loader that takes the file tmp/2008.csv and loads it into database ScyllaDB.


>Its look the same as the command below

What command do you mean? There is nothing below your last message )))

**Comment by SevaMorotskiy** [comment]
It seems your downloaded file with data has been removed. Please follow the instructions from the article:

**$ wget http://stat-computing.org/dataexpo/2009/2008.csv.bz2**
**$ mv 2008.csv.bz2 /tmp**
**$ bzip2 -d /tmp/2008.csv.bz2**

**Comment by dorlaor** [comment]
@tomer-sandler @tzach are you the maintainers of this repository? If so, please review and commit this pul request. We like to blog about it next week

**Comment by tzach** [comment]
@SevaMorotskiy can you please answer my question above?
Can we have a quick interactive session to get this demo working?

**Comment by SevaMorotskiy** [comment]
@tzach I have answered all your 2 questions above (and asked some details about the first one). Could you be so kind to clarify what question do you mean exactly?

**Comment by eyalgutkind** [comment]
@SevaMorotskiy Can you point me to the location in the code in which you are creating the keyspace and table to load the data into the Scylla cluster?
I'm trying to test the code, and get errors on the table creation.
If we have to create the keyspace and table manually, can you advise on the schema/primary key description for the table : spark_scylla.rita

**Comment by tzach** [comment]
@SevaMorotskiy I merged the PR, please handle @eyalgutkind comment ASAP in a **new** PR
Thanks

**Pull request merged**